tests/bearlib/languages/LanguageTest.py

```diff
@@ -46,3 +48,62 @@ def test_invalid_attribute(self):
     def test_attribute_list_empy(self):
         with self.assertRaisesRegex(AttributeError, 'no available attribute'):
             self.lang_unknown.not_an_attribute
+
+
+class LanguagesCollectionDataTypeTest(unittest.TestCase):
+
+    def setUp(self):
+        self.expected_langs = (Language['CPP'], Language['Python'])
+        self.langs = Languages(self.expected_langs)
+
+    def tearDown(self):
+        pass
+
+    def test_collection_data_types(self):
+        self.assertTrue(isinstance(self.langs, abc.Mapping))
+        self.assertTrue(isinstance(self.langs, abc.Iterable))
+
+    def test_languages_apis(self):
+        self.assertEqual(len(self.langs), 2)
+        self.assertEqual(self.langs.keys(), [0, 1])
+        self.assertIn('CPP', self.langs)
+        values = self.langs.values()
+        self.assertIn('CPP', values[0])
+        self.assertIn('Python', values[1])
+        self.assertEqual(len(values), 2)
+        items = self.langs.items()
+        self.assertEqual(len(items), 2)
+        self.assertIn('CPP', self.langs[0])
+        self.assertIn('Python', self.langs[1])
+        reverse = list(reversed(self.langs))
+        self.assertIn('CPP', reverse[1])
+        self.assertIn('Python', reverse[0])
+        self.assertEqual(self.langs, self.expected_langs)
+        langs2 = Languages(self.expected_langs)
+        self.assertTrue(self.langs == langs2)
+        self.assertFalse(self.langs != langs2)
+        self.assertFalse(self.langs == [])
+        cnt = 0
```

Comment:
create a separate test method for this chunk of code, as it is more procedural and detailed, and needs to be read more carefully.